Enforce SELinux in acceptance tests

ekohl opened the pull request:
This attempts to unify SELinux handling in the tests. It moves the package installation to the acceptance spec helper to reduce duplication. It then makes the set_apache_defaults line idempotent and restorecon_apache correctly chained. This works around [PUP-10548] which is that Puppet doesn't reload file contexts within a run. That means it must first create the file(s) and then run restorecon to get correct contexts.

I'm not entirely sure if this will work.

[PUP-10548]: https://tickets.puppetlabs.com/browse/PUP-10548

david22swan commented:
@ekohl Apologies for the late review.
Anyway this look's like a good change to me but was wondering if you had more work that you intended to add to it as it has been left a draft?

ekohl commented:
I'm unable to run tests locally, so I pushed this to see the results. They are red but rotated by now. I'll rebase to see if that's still the case.

ekohl commented:
I've rebased it and split it into two commits. First one that cleans things up (which I think should already be good to merge), then one that makes it enforcing. If the enforcing one fails and we can't quickly figure out why it fails I think we should merge the first commit for now.

david22swan commented:
@ekohl Look's like your getting some failures across the Redhat OSs
Though they don't look fully consistent, some variance across the failures

david22swan commented:
There's not really an easy answer for that. Since the environment is cleaned up at the end of every run, the machines and any log's are all wiped from existence.

Off the top of my head, you could comment out the unnecessary test's and then add a `run_shell` command after the failing one's that cat's said log, allowing you to see it.

ekohl commented:
I split off https://github.com/puppetlabs/puppetlabs-apache/pull/2320 which at least cleans some things up. Let's try to get that merged since I don't have time to finish this for now.

ekohl commented:
Rebased to resolve conflicts. Includes https://github.com/puppetlabs/puppetlabs-apache/pull/2320 so that should be merged first.

david22swan commented:
@ekohl Hey, sorry to bother but just checking in on how this is proceeding so I can update our records?

ekohl commented:
@david22swan I need to do some work on this, but I really struggle to find the time for it. Luckily all the preparation work went in, so I'll rebase this to show that.

LukasAud commented:
Hey @ekohl, are you still interested in working on this project? Perhaps this PR should be closed until work is resumed. Mostly to avoid stale PRs.

ekohl commented:
I don't have time for it right now. Perhaps convert it to an issue so it isn't lost?
